### PR TITLE
fix: scope orphaned-statistics detection to each entry's device slug (#61)

### PR DESCRIPTION
**Pull request by kayl-codes**
## Proposed change
Follow-up to #63. @janusn's multi-entry test of v2.6.1-rc1 (see #61) showed that updating a certificate on one TrueNAS config entry produced **two** duplicate "Orphaned TrueNAS statistics" Repairs issues, even though only one device changed.

Root cause: `_is_truenas_sensor_id` matched a fixed, integration-wide slug (`slugify(DEFAULT_DEVICE_NAME)`, i.e. `"truenas"`). On a multi-entry install, every entry's coordinator scans the *entire* recorder statistics table, so every entry's device-name slug contains that same `"truenas"` substring — meaning each entry's coordinator flagged every *other* entry's orphaned statistics as well, and each entry then raised its own Repairs issue for the same global orphan list.

This PR scopes the match to each config entry's own device-name slug (`slugify(config_entry.data[CONF_NAME])`, e.g. `"truenas_nuc13"` vs `"truenas_x11dpu"`) instead of the fixed constant, so each entry only ever flags statistics that actually belonged to its own device.

This does **not** address the separate `"Handler does not support user"` error @janusn also hit when opening the repair dialog — root cause for that isn't confirmed yet, so it isn't touched here. We'll ask for confirmation whether it recurs after this fix ships.

Fixes/relates to #61.

## Type of change
- [x] Bugfix
- [ ] New feature
- [ ] Code quality improvements to existing code or addition of tests
- [ ] Documentation

## Additional information
- New regression test `test_async_detect_orphaned_statistics_ignores_other_entrys_device` reproduces the two-entry scenario from @janusn's log.
- `_is_truenas_sensor_id` now takes `device_slug` as a required parameter instead of reading a module-level constant.

## Checklist
- [x] The code change is tested and works locally.
- [x] The code has been formatted and linted using Ruff.
- [x] Tests have been added to verify that the new code works.
- [ ] Documentation added/updated if required.
- [ ] Tested against the latest Home Assistant version.

## Summary by Sourcery

Scope orphaned statistics detection to each TrueNAS config entry’s device slug to avoid cross-entry conflicts on multi-entry installs.

Bug Fixes:
- Ensure orphaned statistics detection only matches statistics belonging to the current config entry’s device slug instead of a global slug, preventing duplicate Repairs issues across entries.

Tests:
- Add regression tests covering per-entry device-slug matching, empty-device-slug rejection, and multi-entry scenarios for orphaned statistics detection.

### CODE REVIEW

File: custom_components/truenas_ce/coordinator.py [hunk 1]
```diff
@@ -320,34 +319,29 @@ def _first_ipv4(aliases: Any) -> str:
...
-    underscore so it can never appear whole inside an underscore-split token
-    (this silently broke orphan detection from the 2.0.0 CE rename until found
-    while adding this module's test suite). Tying the match to the device name
-    instead of ``LEGACY_DOMAIN`` means this keeps working unchanged even if the
-    legacy-migration code/domain is ever removed (e.g. a future HA Core
-    submission dropping the "_ce" suffix).
+    per-entry device-name slug as a substring of the id's remainder after
+    ``sensor.`` rather than as an exact token or fixed prefix, so every
+    orphaned variant is caught.
+
```
[the remark this answers is not in the document]
Comment (kayl-codes) [reply custom_components/truenas_ce/coordinator.py:331]: Fixed in f7a71ad: `_is_truenas_sensor_id` now returns `False` outright when `device_slug` is falsy, and the slug is cached once in `__init__` per the other overall comment rather than recomputed every poll.

**Comment by kayl-codes** [comment]
Addressing the two overall comments from the Sourcery review:

> In `async_detect_orphaned_statistics`, using `slugify(self.config_entry.data[CONF_NAME])` assumes `CONF_NAME` is always present and non-empty; consider a defensive fallback...

`CONF_NAME` is `vol.Required` in the config flow schema (with a default), so it's always present — a missing-key guard would be dead code and inconsistent with every other call site in this codebase (`entity.py`, `button.py`, `sensor.py`, `__init__.py` all access `config_entry.data[CONF_NAME]` directly). What *is* possible under the schema is an **empty string** (plain `str` validator, no min-length), which is the real risk here — an empty slug would match every statistic id as a substring. Fixed in f7a71ad with an explicit `if not device_slug: return False` guard in `_is_truenas_sensor_id`, covered by `test_is_truenas_sensor_id_rejects_empty_device_slug`.

> Now that `_is_truenas_sensor_id` is per-entry, you are recomputing the slug inside `async_detect_orphaned_statistics` on every call; consider storing the slug on the coordinator...

Fixed in f7a71ad: the slug is now computed once in `TrueNASCoordinator.__init__` as `self._device_slug` and reused on every poll. A config-entry rename goes through a full entry reload (new coordinator instance via `async_update_reload_and_abort`), so caching it at construction time never goes stale.

**Comment by kayl-codes** [comment]
@sourcery-ai review
